iris-mpc-cpu/src/execution/hawk_main/reset.rs

```diff
@@ -0,0 +1,38 @@
+use super::{
+    search::{self, SearchParams, SearchQueries, SearchResults},
+    BothEyes, HawkActor, HawkRequest, HawkSessionRef, LEFT,
+};
+pub use crate::hawkers::aby3::aby3_store::VectorId;
+use eyre::Result;
+
+pub struct ResetRequests {
+    pub vector_ids: Vec<VectorId>,
+    pub queries: SearchQueries,
+}
+
+pub struct ResetPlan {
+    pub vector_ids: Vec<VectorId>,
+    pub search_results: SearchResults,
+}
+
+pub async fn search_to_reset(
+    hawk_actor: &mut HawkActor,
+    sessions: &BothEyes<Vec<HawkSessionRef>>,
+    request: &HawkRequest,
+) -> Result<ResetPlan> {
+    // Get the reset updates from the request.
+    let updates = {
+        let store = hawk_actor.iris_store[LEFT].read().await;
```

Comment:
Triggered by quick confusion of only `LEFT` usage here, is there an actual need to have two `iris_store` Arcs for left/right in the `HawkActor`? They point to the same RwLocked IrisStore anyway, right?

Reply:
This is to obtain the full vector IDs from indices, which should be the same in both stores. The two stores are otherwise used independently.

Reply:
Added a comment.